metaflow/plugins/logs_cli.py

```diff
@@ -0,0 +1,306 @@
+                elif maybe_old_style:
+                    # if they are not available, we may be looking at
+                    # a legacy run (unless we have seen new-style data already
+                    # for another stream). This return an empty string if
+                    # nothing is found
+                    log = ds.load_log_legacy(stream, attempt_override=attempt)
+                    if log and timestamps:
+                        raise CommandException(
+                            "We can't show --timestamps for old runs. Sorry!"
+                        )
+                    echo_unicode(log, nl=False)
+    else:
+        raise CommandException(
+            "No Tasks found at the given path -- "
+            "either none exist or none have started yet"
+        )
+
+
+@logs.command(
+    help="Scrub stdout/stderr produced by a task or all tasks in a step. "
+    "The format for input-path is either <run_id>/<step_name> or "
+    "<run_id>/<step_name>/<task_id>."
+)
+@click.argument("input-path")
+@click.option(
+    "--stdout/--no-stdout",
+    default=False,
+    show_default=True,
+    help="Scrub stdout of the task.",
+)
+@click.option(
+    "--stderr/--no-stderr",
+    default=False,
+    show_default=True,
+    help="Scrub stderr of the task.",
+)
+@click.option(
+    "--both/--no-both",
+    default=True,
+    show_default=True,
+    help="Scrub both stdout and stderr of the task.",
+)
+@click.option(
+    "--attempt",
+    default=None,
+    type=int,
+    show_default=False,
+    help="Attempt number of a task to scrub, defaults to the latest attempt.",
+)
+@click.pass_obj
+def scrub(obj, input_path, stdout=None, stderr=None, both=None, attempt=None):
+    types = set()
+    if stdout:
+        types.add("stdout")
+        both = False
+    if stderr:
+        types.add("stderr")
+        both = False
+    if both:
+        types.update(("stdout", "stderr"))
+
+    streams = list(sorted(types, reverse=True))
+
+    # Pathspec can either be run_id/step_name or run_id/step_name/task_id.
+    parts = input_path.split("/")
+    if len(parts) == 2:
+        run_id, step_name = parts
+        task_id = None
+    elif len(parts) == 3:
+        run_id, step_name, task_id = parts
+    else:
+        raise CommandException(
+            "input_path should either be run_id/step_name "
+            "or run_id/step_name/task_id"
+        )
+
+    if task_id:
+        ds_list = [
+            TaskDataStore(
+                obj.flow_datastore,
+                run_id=run_id,
+                step_name=step_name,
+                task_id=task_id,
+                attempt=attempt,
+                mode="d",
+            )
+        ]
+    else:
+        ds_list = obj.flow_datastore.get_latest_task_datastores(
+            run_id=run_id, steps=[step_name], attempt=attempt, mode="d"
+        )
+
+    if ds_list:
+        for ds in ds_list:
+            for stream in streams:
+                ds.scrub_logs(LOG_SOURCES, stream)
```

Comment:
This is a bit annoying in the case some logs get scrubbed but not others. I would do this in a try-catch and do as much as possible but report partial failures (maybe with echo_always?)